torchvision/transforms/functional_tensor.py

```diff
@@ -50,15 +46,31 @@ def crop(img, top, left, height, width):
     return img[..., top:top + height, left:left + width]
 
 
+def rgb_to_grayscale(img):
+    """Convert the given RGB Image Tensor to Grayscale.
+
+    Args
+        img (Tensor): Image to be converted to Grayscale in the form [C, H, W].
+        Tensor: Grayscale image.
+
+    For RGB to Grayscale conversion, ITU-R 601-2 luma transform is performed which
+    is L = R * 0.2989 + G * 0.5870 + B * 0.1140
+    """
+    if img.size()[0] != 3:
+        raise TypeError('Input Image does not contain 3 Channels')
+
+    device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
```

Comment:
Please keep this part as before. This is an implicit cast which can lead to very unexpected results.
For example, if a user passes a CPU tensor, but has a CUDA device available, this will be returning a CUDA tensor, which is definitely not expected, and is the reason why one of the tests is failing.